src/configuration/default_config.rs

```diff
@@ -0,0 +1,74 @@
+/// What environment are we running in?
+#[derive(Debug, Clone)]
+pub enum Environment {
+    Development,
+    Staging,
+    Production,
+}
+
+/// Holds the default configuration for the App
+#[derive(Debug, Clone)]
+pub struct Configuration {
```

Comment:
I wanted the `Configuration` type to be the defaults supplied by the framework, while the users could use the store to store any other types that they would want.

But given that they can pass their own `Data`, is it even necessary?

Reply:
Plus, I feel that a single `Config` struct would be more ergonomic to use rather than individual configuration items. I mean you would have to go through the documentation to figure out what the individual items are.

Reply:
Well, I was thinking about overriding individual items; but it is unlikely to happen since this is a global configuration. `Store` (originally `Configuration`) is written with overriding in mind, so now I think it *might* be better if global configuration is stored elsewhere, in the form of typemap or whatever.

Reply:
I meant, global configuration can be overridden *per endpoint* with current API.

```rust
let config_global = Configuration {
    env: Environment::Development,
    ..Configuration::default()
};
app.config(config_global);
// `foo` will see development environment
app.at("foo").get(foo);

let config_overridden = Configuration {
    env: Environment::Production,
    ..Configuration::default()
};
// `bar` will see production environment
app.at("bar").get(bar).config(config_overridden);
```

Well, instead of my suggestion (a new field for global configuration), we can just make sure library consumers to apply global configuration only to the top-level router, by writing about it in the documentation. Let's go with the latter (your current code with additional documentation), and revisit this later.

Reply:
Sure thing! How will we go about it though? Will you merge my PR here into your branch and will that update your PR on the tide repo?